orderbook/src/database.rs

```diff
@@ -11,16 +15,148 @@ pub struct Trade {
     pub fee_amount: U256,
 }
 
-/// Abstraction over the database we use to store persistent data like orders and trade events.
-#[async_trait::async_trait]
-pub trait Database {
-    async fn block_number_of_most_recent_trade(&self) -> Result<u64>;
```

Comment:
Changed from trait to struct for simplicity. We can still turn it back into a trait if we need to but I don't want to do it proactively.